### PR TITLE
[Feature] Admin module

### PR DESCRIPTION
**Pull request by Ghalko**
## Proposal
Adding a general purpose admin module with the some spam routes, views, and digging into routes for templates.

This is meant to be a pretty barebones implementation, for use by a wider team looking to add apps to the Admin module in order to support user, pre-reg, and spam administration.

## Repercussions
Another django app inside of a flask project.

### CODE REVIEW

File: admin/base/settings/defaults.py [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+# Application definition
+
+INSTALLED_APPS = (
+    'django.contrib.admin',
+    'django.contrib.auth',
+    'django.contrib.contenttypes',
+    'django.contrib.staticfiles',
+
+    # 3rd party
+    'corsheaders',
```
Comment (sloria) [inline admin/base/settings/defaults.py]: Do you need CORS for this app?

File: admin/base/settings/defaults.py [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+TEMPLATES = [
+    {
+        'BACKEND': 'django.template.backends.django.DjangoTemplates',
+        'DIRS': [os.path.join(BASE_DIR, 'templates')],
+        'APP_DIRS': True
+    }]
+
+
+ROOT_URLCONF = 'admin.base.urls'
+WSGI_APPLICATION = 'api.base.wsgi.application'
```
Comment (sloria) [inline admin/base/settings/defaults.py]: This should point to the admin app's wsgi app.

**Comment by sloria** [comment]
Rather than having multiple `manage.py` files, we should be able to parametrize a single `manage.py` file with the `DJANGO_SETTINGS_MODULE` envvar.

```python
DJANGO_SETTINGS_MODULE='admin.base.settings' python manage.py runserver
```

**Comment by sloria** [comment]
Pass finished.

File: admin/base/settings/defaults.py (hunk 1 above)
Comment (Ghalko) [inline admin/base/settings/defaults.py]: Probably not. Removed.

**Comment by Ghalko** [comment]
So rather than using the normal `setdefault` it would just be `set`?

File: admin/base/settings/defaults.py [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+)
+
+# TODO: Are there more granular ways to configure reporting specifically related to the API?
+RAVEN_CONFIG = {
+    'dsn': osf_settings.SENTRY_DSN
+}
+
+# Settings related to CORS Headers addon: allow API to receive authenticated requests from OSF
+# CORS plugin only matches based on "netloc" part of URL, so as workaround we add that to the list
+CORS_ORIGIN_ALLOW_ALL = False
```
Comment (sloria) [inline admin/base/settings/defaults.py]: Remove these and any other settings that aren't used.

**Comment by sloria** [comment]
Python modules cannot contain dashes; change `pre-reg` -> `pre_reg`

**Comment by Ghalko** [comment]
Stopped the build for the pre-reg to pre_reg change.

**Comment by sloria** [comment]
Pass finished.

**Comment by sloria** [comment]
Merged into `feature/admin-app`. Please send future PRs to that branch.

**Pull request closed without merging**